compiler/qsc_formatter/src/formatter.rs

```diff
@@ -490,29 +536,51 @@ impl<'a> Formatter<'a> {
         }
     }
 
+    fn update_import_export_state(&mut self, left_kind: &ConcreteTokenKind) {
+        use qsc_frontend::keyword::Keyword;
+        use ConcreteTokenKind::*;
+        use TokenKind::*;
+
+        match left_kind {
+            Comment => {
+                // Comments don't update state
+            }
+            Syntax(Keyword(Keyword::Import | Keyword::Export)) => {
+                self.import_export_state = ImportExportState::HandlingImportExportStatement;
+            }
+            Syntax(Semi) => {
+                self.import_export_state = ImportExportState::NoState;
+            }
+            _ => (),
+        }
+    }
+
     /// Updates the type_param_state of the FormatterState based
-    /// on the left and right token kinds. Returns the delimiter
-    /// state of the left and right tokens.
+    /// on the left and right token kinds.
     fn update_type_param_state(
         &mut self,
         left_kind: &ConcreteTokenKind,
         right_kind: &ConcreteTokenKind,
-    ) -> (Delimiter, Delimiter) {
+    ) {
         use qsc_frontend::keyword::Keyword;
         use ConcreteTokenKind::*;
         use TokenKind::*;
 
-        // Save the left token's status as a delimiter before updating the delimiter state
-        let left_delim_state = Delimiter::from_concrete_toke_kind(left_kind, self.type_param_state);
-
-        // If we are leaving a type param list, reset the state
-        if matches!(left_kind, Syntax(Gt))
-            && matches!(
-                self.type_param_state,
-                TypeParameterListState::InTypeParamList
-            )
-        {
-            self.type_param_state = TypeParameterListState::NoState
+        // Based on the left-hand-side token, we determine if:
```

Comment:
This comment is no longer correct, now that the import_export logic was separated out. I think this chunk of code could go back to the way it was before, unless you think the match statement is better. Either way, the comment should be corrected.